Ruby on Rails Deployment: Fix grammatical errors (#25113)
TheOdinProject/curriculum: ruby_on_rails/rails_basics/deployment.md

### Introduction

> Before continuing our web development journey, we must address one important task—showcasing our hard work to the world.

Reviewer: Are you sure this is correct? these don't seem like two words that can be hyphenated together.

Author: My change switched it from a hyphen (-) to an em dash (—). Using an em dash in this context serves to define the final clause (i.e., "showcasing …") as a summarizing/expanding statement. See [Merriam-Webster](https://www.merriam-webster.com/words-at-play/em-dash-en-dash-how-to-use) for more details.

If the em dash feels wrong to you, another option would be to use a colon:
"… we must address one important task: showcasing our hard work …"

If you don't like either of those options, I think your original form was perfectly readable as it was. Grammar rules help provide consistency, but readability is all that really matters.

Reviewer: ah sorry, my mistake. I really like your colon suggestion. Can we go with that please?

This additional tech prohibits us from using GitHub Pages for hosting our Ruby on Rails apps. GitHub Pages cannot run Ruby code and doesn't have database services we can use.

Luckily, many hosting providers do offer everything we need. They range from the big and complex cloud providers like [AWS](https://aws.amazon.com/), [Google Cloud](https://cloud.google.com/) and [Microsoft Azure](https://azure.microsoft.com/) to the more beginner-friendly platform as a service (PaaS) providers like [Heroku](https://www.heroku.com/), [Railway](https://railway.app/) and [Fly.io](https://fly.io/). We will be focusing on and utilizing these latter providers in this lesson.

### What is a PaaS?
Platform as a Service is a specific kind of hosting provider. The most important thing to know about them is they are much easier to use and more approachable for beginners than other hosting providers. They manage many of the low-level nitty-gritty details with the underlying server infrastructure. Allowing us as developers to focus more of our time on building our applications instead of configuring and managing the servers they run on.

> Taking our landlord metaphor a little further, a PaaS platform is like having a landlord who takes care of all the utilities, building maintenance and security. While you, the developer, focus on furnishing, decorating and living in the space.

Reviewer: Plural "focuses" reads better to me here.

Author: It's easier to see the subject-verb agreement if we take out "the developer":
"While you focus on furnishing …"

Reviewer: I see what you mean, great idea!

The cool thing about Rails is that you can always fire up more instances of your application if you get too much traffic and users have to wait for their requests to be filled.

For most of your apps, one instance is plenty. You can support a lot of traffic using just a single instance. Many of the PaaS providers we will recommend later in this lesson give you your first one for free.

#### Databases
The second most important thing PaaS providers give you is databases. They make it easy to spin up a new database for each app by doing all the setup and configuration for you.

Many providers even manage the database for you by setting up automatic backups, ensuring the database is constantly updated with the latest critical security patches and ongoing maintenance that keeps your databases up and running smoothly.

The peace of mind this affords you can't be overstated. You never want to be in a position where you're waking at 4 a.m. to multiple alerts. Your database is on the fritz because of some security patch you forgot to apply, and there are no backups to fall back on.

> With Rails, we will be using PostgreSQL, a popular open-source database. Some PaaS providers like Heroku will automatically create a PostgreSQL database for our application when we first deploy. Others, like Fly.io, will have a few more manual steps involved, but still beat having to set up a database from scratch.

Reviewer: Plural "beats" reads better to me here.

Author: Here's a third idea:
"… but it still beats having to set up …"

Here, the pronoun's antecedent isn't intended to be "others" (i.e., other PaaS providers) but instead the whole situation (i.e., having to do a few manual steps still beats setting up a database from scratch). I think this may be more inline with your original intention.

Reviewer: Sounds perfect, lets do it

So if you're getting a 500 error, you can open up the logs and watch them closely as you refresh the page on your app in the browser to reproduce the error. This will either tell you exactly what the problem is or generate some leads you can use to dig further.

> If the logs tell you the application crashes before it even boots and do not give you anything more to go on, the next tool to reach for is the Rails console. The Rails console boots your app, so just opening it in production can produce a better error message that will tell you exactly why the app is crashing and what part of the app is causing it.

Reviewer: Do doesn't read well here. 

perhaps something like this instead:
```suggestion
If the logs tell you the application crashes before it even boots and gives you nothing else to go on, the next tool to reach for is the Rails console. The Rails console boots your app, so just opening it in production can produce a better error message that will tell you exactly why the app is crashing and what part of the app is causing it.
```

Author: I like your suggestion. I think it'd need to be "give" instead of "gives" since we're treating "the logs" as plural. It's clearer if I remove the first part of the phrase:
"If the logs give you nothing else to do on …"

Reviewer: Sounds good, thanks @rhodes-david 💪
